packages/frontend/src/view/Form/ObjectArrayWithSelect/ObjectArrayWithSelect.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import clsx from 'clsx';
+import { Controller } from 'react-hook-form';
+
+import { Remove } from '../../../icons';
+import { ObjectArrayWithSelectForm as Props } from '../../../types';
+import { Col, IconButton, Row, Input, Tooltip, Select } from '../../../ui';
+import { useStyles } from './useStyles';
+
+export const ObjectArrayWithSelect: React.FC<Props> = ({
+  onRemove,
+  register,
+  errors,
+  fields,
+  name,
+  options,
+  label = 'key',
+  helperText = '',
+  control,
+  selectLabel = 'select',
+}) => {
+  const { objectField, lastField, objectRow } = useStyles();
+
+  const lastObjectField = clsx(objectField, lastField);
+
+  return (
+    <>
+      {fields.length > 0 &&
+        fields.map((field, idx) => {
+          const remove = (
+            <Col xs={1} className={lastObjectField}>
+              <Tooltip title="Remove field" placement="left">
+                <IconButton onClick={() => onRemove(idx)}>
+                  <Remove />
+                </IconButton>
+              </Tooltip>
+            </Col>
+          );
```

Comment:
We can use this component directly at the place. We don't need to create extra `const`.